File: Euclid.Plugin.Example/PluginExample.cs
using Euclid.Game;
using log4net;
using System.Reflection;

namespace Plugin.Example
{
    public class PluginExample : IPlugin
    {
        #region Fields

        private static readonly ILog log = LogManager.GetLogger(MethodBase.GetCurrentMethod().DeclaringType);

        #endregion

        #region Public methods

        public void onEnable()
        {
            log.Info("Enabling Plugin Example");
        }

        public void onDisable()
        {
            log.Info("Disposing Plugin Example");
        }

        #endregion
    }
}

File: euclid-plugin-example/src/main/java/plugin/example/PluginExample.java
package plugin.example;

import euclid.game.plugins.IPlugin;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class PluginExample implements IPlugin {

    private static final Logger log = LogManager.getLogger(PluginExample.class);

    @Override
    public void onEnable() {
        log.info("Enabling Plugin Example");
    }

    @Override
    public void onDisable() {
        log.info("Disposing Plugin Example");
    }
}
